File: leetcode/minimum-operations-to-make-the-array-increasing.cpp
class Solution {
  public:
    int minOperations(vector<int> nums){
        int sum = 0;
        int i = 1;
        std::for_each(std::begin(nums) + 1,
                      std::end(nums),
                      [&nums, &sum, &i](int v)
                      { sum += std::max(0, nums[i - 1] - v + 1);
                        nums[i] = std::max(nums[i], nums[i - 1] + 1);
                        i++;
                      });
        return sum;
    }
};

File: leetcode/minimum-operations-to-make-the-array-increasing.py
from typing import List


class Solution:
    def minOperations(self, nums: List[int]) -> int:
        s = 0
        for i in range(1, len(nums)):
            d = max(0, 1 + nums[i - 1] - nums[i])
            nums[i] = nums[i] + d
            s += d
        return s
